project3/participant/input_parser/input_parser.h

```diff
@@ -46,43 +46,18 @@ class InputParser {
   google::protobuf::Message* CreateReq();
 
  private:
-  /**
-   * creates a register message
-   */
-  pub_sub_messages::Register* CreateRegMsg();
-
-  /**
-   * creates a reregister message
-   */
-  pub_sub_messages::Deregister* CreateDeregMsg();
-
-  /**
-   * creates a disconnect message
-   */
-  pub_sub_messages::Disconnect* CreateDisconMsg();
-
-  /**
-   * creates a reconnect message
-   */
-  pub_sub_messages::Reconnect* CreateReconMsg();
-
-  /**
-   * creates a multicast send message
-   */
-  pub_sub_messages::SendMulticast* CreateMsendMsg();
-
   const std::map<std::string, MsgType> commands_ = {{"register", REG},
                                                     {"deregister", DEREG},
                                                     {"disconnect", DISCON},
                                                     {"reconnect", RECON},
                                                     {"msend", MSEND}};
 
   /// messages having been parsed from input
-  pub_sub_messages::Register* reg_msg_;
-  pub_sub_messages::Deregister* dereg_msg_;
-  pub_sub_messages::Disconnect* discon_msg_;
-  pub_sub_messages::Reconnect* recon_msg_;
-  pub_sub_messages::SendMulticast* msend_msg_;
+  pub_sub_messages::Register reg_msg_;
```

Comment:
Looks good! Just for reference, you should be able to initialize these members just by appending `{}` to the definition here. Then there will be no need to do it manually in the constructor. (Even if you don't, it should still work, but I think `clang-tidy` yells at you.)

Reply:
Seems it didn't really yell at me interestingly enough. I do like how that cleaned up some of the initialization.